### ISSUE TITLE
Webpacker compatibility

### ISSUE THREAD

**Issue by rossta** [rossta/serviceworker-rails#41]
The `webpacker` gem compiles "packs" to a separate subdirectory in `public` by default, e.g. `public/packs`. There should be some way to proxy serviceworker requests to the appropriate pack when `webpacker` is used.

**Comment by mikecx** [comment]
Would love to see this 👍

**Comment by toxin20** [comment]
Could you give an example how this would work for an `application.js` pack from webpacker? Webpacker spits out the fingerprinted `application.js` and makes it available under `/packs/application-69ae1a92cae9e2edf33e.js` or similar. 

How can I reference this pack in my `serviceworker.js.erb` so the file can be cached? In your README you have an webpacker example in the initializer but I'm not how sure how it relates to my problem as there is no fingerprint in the filenames:

```
# maps to serviceworker "pack" compiled by Webpacker
match "/webpack-serviceworker.js" => "serviceworker.js", pack: true
```

**Comment by rossta** [comment]
If you're creating a serviceworker in Webpack, you'd want to create a separate entry file, such as `app/javascript/packs/serviceworker.js`. When Webpacker compiles this file, it will output along side your other packs as `/public/packs/serviceworker-somedigest.js`.

The thing is, we don't want the serviceworker to be cached nor have a fingerprinted url, nor do we want it active on the path `/packs`. 

Now, with the following configuration:

```
match "/serviceworker.js" => "serviceworker.js", pack: true
# or simply, in this case
match "/serviceworker.js", pack: true 
```

...you'll be able to map requests from `/serviceworker.js` to the currently compiled serviceworker "pack", i.e., the request is proxied to `/packs/serviceworker-somedigest.js`. The browser should only see the non-digest url and will be able to detect the new serviceworker when it changes.

Does that answer your question?
